### DISCUSSION TITLE
Contributing to the repo

### DISCUSSION THREAD

**Discussion by howardnguyen714** [xblax/flashforge_ad5m_klipper_mod#227] [Q&A]
I'm interested in contributing to this project, but I'm not too sure where to start. Can someone help point me in the right direction? My experience is mostly in React, but I have a little Python experience as well. Thanks!

**Comment by KaruroChori** [comment]
This repository makes no use of JS, let alone react. There is a bit of python involved since it is related to klipper, but it is marginal. Also, I don't have the feeling there is much left to do in the first place.
The firmware works, and many further developments are locked behind hardware limitations.

If you don't have a problem with this specific printer already, why would you want to contribute?

**Reply by consp** [reply]
> don't have the feeling there is much left to do in the first place

Some guppyscreen gui work but you'd need C/C++ for that, unless someone wants to tackle the python memory issues but that's not a small task (and I would avoid it unless you are _VERY_ familiar with it)

**Reply by howardnguyen714** [reply]
I'll be getting this printer soon and figured I take the opportunity to learn stuff that's new to me while contributing to something that would help me and others. If I find some time I'll pick up C/C++ and try to improve the gui :)

**Reply by consp** [reply]
When you are, let me know and I'll write a small howto on building guppyscreen out-of-tree. Though you can use the emulation for some non-ad5m specific things (see the guppyscreen original repo for that)
